android/src/main/java/com/swmansion/reanimated/keyboard/Keyboard.java

```diff
@@ -20,15 +21,31 @@ public int getHeight() {
   public void updateHeight(WindowInsetsCompat insets) {
     int contentBottomInset = insets.getInsets(CONTENT_TYPE_MASK).bottom;
     int systemBarBottomInset = insets.getInsets(SYSTEM_BAR_TYPE_MASK).bottom;
-    int keyboardHeightDip = contentBottomInset - systemBarBottomInset;
-    mHeight = (int) PixelUtil.toDIPFromPixel(Math.max(0, keyboardHeightDip));
+    int keyboardHeightDip = Math.max(0, contentBottomInset - systemBarBottomInset);
+    if (keyboardHeightDip == 0 && mState == KeyboardState.OPEN) {
+      /*
+      When the keyboard is being canceling, for one frame the insets show a keyboard height of 0,
+      causing a jump of the keyboard. We can avoid it by ignoring that frame and calling
+      the listeners on the following frame.
+      */
+      return;
+    }
+    mHeight = (int) PixelUtil.toDIPFromPixel(keyboardHeightDip);
```

Comment:
This change looks similar to the one from #5742, is it supposed to be in this PR?

Reply:
This change may seem minor, but it's crucial to implement in this PR to ensure correct state management.

It sets the keyboard height to 0, so when the next animation begins, in the `onAnimationStart` method, we will switch the state to `opening` instead of `closing`.